include/dxc/HLSL/HLOperations.h

```diff
@@ -389,6 +390,18 @@ llvm::Function *GetOrCreateHLFunction(llvm::Module &M,
                                       llvm::StringRef *fnName,
                                       unsigned opcode);
 
+llvm::Function *GetOrCreateHLFunction(llvm::Module &M,
+                                      llvm::FunctionType *funcTy,
+                                      HLOpcodeGroup group, unsigned opcode,
+                                      const llvm::AttributeSet &attribs);
+llvm::Function *GetOrCreateHLFunction(llvm::Module &M,
+                                      llvm::FunctionType *funcTy,
+                                      HLOpcodeGroup group,
+                                      llvm::StringRef *groupName,
+                                      llvm::StringRef *fnName,
+                                      unsigned opcode,
+                                      const llvm::AttributeSet &attribs);
+
```

Comment:
These variants allow providing attribs from a function that this will replace to propagate them. This is strictly needed only for wave sensitivity, but will also copy over readnone and readonly. These don't currently see any use after codegen is done, but copying them over is cheap, matches expectations better, and causes no harm.